Base/Python/slicer/util.py

```diff
@@ -2498,14 +2498,40 @@ def arrayFromTableColumnModified(tableNode, columnName):
 
 
 def updateTableFromArray(tableNode, narrays, columnNames=None):
-    """Set values in a table node from a numpy array.
-
-    :param columnNames: may contain a string or list of strings that will be used as column name(s).
-    :raises ValueError: in case of failure
-
-    Values are copied, therefore if the numpy array  is modified after calling this method,
-    values in the table node will not change.
-    All previous content of the table is deleted.
+    """Set values in a table node from a NumPy array or an array-like object (list/tuple of NumPy arrays).
+
+    This function can handle:
+
+      - **Structured (record) array** with named fields. Each field becomes a separate column.
+        If ``columnNames`` is not provided, the field names in the dtype are used.
+      - **1D NumPy array**, which is added as a single column.
+      - **2D NumPy array**, where each column in the array is added as a separate column in the
+        table node (note that the array is transposed).
+      - **List/tuple** of 1D NumPy arrays, which are each added as separate columns.
+
+    :param tableNode: The table node to be updated. If ``None``, a new ``vtkMRMLTableNode`` is
+                      created and added to the scene.
+    :type tableNode: vtkMRMLTableNode or None
```

Comment:
shouldn't we use type annotations instead?

Reply:
Good point. This will be done in a follow-up pull request:
* https://github.com/Slicer/Slicer/issues/8226